src/components/PopupWithForm.js

```diff
@@ -30,6 +32,12 @@ export class PopupWithForm extends Popup {
     this._popupForm.reset();
     super.close();
   }
-}
```

Comment:
Needs correction: (line 27) Сlose modal in the case of successful api response in `then` block in `index.js`, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again.